library/common/extensions/filters/http/platform_bridge/filter.h

```diff
@@ -160,6 +162,7 @@ class PlatformBridgeFilter final : public Http::PassThroughFilter,
   envoy_http_filter_callbacks platform_request_callbacks_{};
   envoy_http_filter_callbacks platform_response_callbacks_{};
   bool error_response_{};
+  bool alive_{true};
```

Comment:
This is fine and should work. Just for illustrative/educational purposes, I was suggesting the following:
1) `std::shared_ptr<BridgeFilterHandle> still_alive_`;
2) `BridgeFilterHandle` contains a reference to the filter.
3) Return a weak `BridgeFilterHandle` everywhere you now return a weak filter using `weak_from_this`.
4) Locking the weak `BridgeFilterHandle` is sufficient to understand liveiness if you `still_alive.reset()` on `onDestroy()`.

The mild benefit here is I think it's a bit more intentional what you are doing with less boolean logic, but again this is fine.